.npmignore

```diff
@@ -0,0 +1,38 @@
+.DS_Store
+.git/
+.github/
+.gitignore
+.gitmodules
+.clang-*
+.cmake-format
+
+.vscode/
+
+patches/
+pkgconfig/
+scripts/
+test*/
+build/
+ci/
+docs/
+example*/
+
+xpacks/
+node_modules/
+
+*.tgz
+
+README-*
+LICENSE-*
+CONTRIBUTING**
+
+Doxyfile
+Makefile
+img/
+include/
+logo.drawio.svg
+
+.gitignore
+
+# End
```

Comment:
This .npmignore file seems quite large. Wouldn't it be more concise to exclude everything and only include the needed files?

Reply:
It might be more concise, but I don't know if it is supported, I always used this file as the name implies, to define files/folders that are **not** to be included in the package.

If you have experience with such negated configurations, please suggest a solution.

Reply:
According to the documentation, npm supports the same syntax as .gitignore files: <https://docs.npmjs.com/cli/v9/using-npm/developers?v=true#keeping-files-out-of-your-package>

I've never used npm, but in a .gitignore you can do something like:

```gitignore
!/*
/doxygen-awesome*
```

Reply:
Yeah maybe it doesn't understand the leading slash. What if you drop it? I.e. `!*` and `doxygen-awesome*`

Reply:
Oh, I just noticed that I've made a typo in my first comment. It shouldn't be `/!*` but `!/*`

Reply:
The output above was generated with that `.npmignore`, so I guess it is ok. 

If in the future you'll add new files/folders to the project which do not need to be propagated in the npm package, you can always add more entries to this file.

Reply:
For the record, my suggestion above was wrong; I wrote the opposite of what I meant 😅️

It should've been:

```gitignore
*
!doxygen-awesome*
```

i.e. ignore everything except for files starting with the `doxygen-awesome` name. This should require less maintenance than manually enumerating all the files that have to be excluded.

Reply:
I confirm that this new syntax seems fine, it produced the following output:

```
> npm pack

npm notice 
npm notice 📦  @jothepro/doxygen-awesome-css@2.2.1-pre
npm notice === Tarball Contents === 
npm notice 1.1kB  LICENSE                                         
npm notice 5.7kB  README.md                                       
npm notice 8.5kB  doxygen-awesome-darkmode-toggle.js              
npm notice 4.0kB  doxygen-awesome-fragment-copy-button.js         
npm notice 3.2kB  doxygen-awesome-interactive-toc.js              
npm notice 2.4kB  doxygen-awesome-paragraph-link.js               
npm notice 1.5kB  doxygen-awesome-sidebar-only-darkmode-toggle.css
npm notice 3.2kB  doxygen-awesome-sidebar-only.css                
npm notice 3.0kB  doxygen-awesome-tabs.js                         
npm notice 63.0kB doxygen-awesome.css                             
npm notice 944B   package.json                                    
npm notice === Tarball Details === 
npm notice name:          @jothepro/doxygen-awesome-css              
npm notice version:       2.2.1-pre                                  
npm notice filename:      @jothepro/doxygen-awesome-css-2.2.1-pre.tgz
npm notice package size:  18.9 kB                                    
npm notice unpacked size: 96.6 kB                                    
npm notice shasum:        2d9ebc7f7a6eb966e5008c4c79556c458e591172   
npm notice integrity:     sha512-j485njHztxDDC[...]tA6Lpq2Vu7XSA==   
npm notice total files:   11                                         
npm notice 
jothepro-doxygen-awesome-css-2.2.1-pre.tgz
```

If this is the expected content to be published, you can leave it like this, otherwise you can further edit it.